### PR TITLE
#17262 - Changing password should respect AllowNoPassword

### PR DESCRIPTION
**Pull request by RamiRima**
#17262

### CODE REVIEW

**Review by williamdes — CHANGES_REQUESTED** [review]
Please use a translation string in the controller file: https://github.com/phpmyadmin/phpmyadmin/blob/master/libraries/classes/Controllers/JavaScriptMessagesController.php#L40

By the way there is a typo in the sentence

File: libraries/classes/Controllers/JavaScriptMessagesController.php [hunk 1]
```diff
@@ -41,6 +41,9 @@ private function setMessages(): array
     {
         return [
             /* For confirmations */
+            'strAllowNoPassowrdIsFalse' => __(
+                'You are trying to set this account to log in without a password but the configuration directive AllowNoPassword is false. If you procced, the account will not be able to log in through phpMyAdmin. Do you wish to procced ? '
```
Comment (liviuconcioiu) [inline libraries/classes/Controllers/JavaScriptMessagesController.php:45]: `You are trying to set this account to log in without a password but the configuration directive AllowNoPassword is false. If you proceed, the account will not be able to log in through phpMyAdmin. Do you wish to proceed?`

**Review by liviuconcioiu — CHANGES_REQUESTED** [review]

File: libraries/classes/Controllers/JavaScriptMessagesController.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ private function setMessages(): array
         return [
             /* For confirmations */
             'strAllowNoPassowrdIsFalse' => __(
-                'You are trying to set this account to log in without a password but the configuration directive AllowNoPassword is false. If you procced, the account will not be able to log in through phpMyAdmin. Do you wish to procced ? '
+            'You are trying to set this account to log in without a password but the configuration directive AllowNoPassword is false. If you proceed, the account will not be able to log in through phpMyAdmin. Do you wish to proceed?'
```
Comment (liviuconcioiu) [inline libraries/classes/Controllers/JavaScriptMessagesController.php:45]: Can you remove the white space at the end?

**Review by liviuconcioiu — CHANGES_REQUESTED** [review]

File: js/src/functions.js [hunk 3]
```diff
@@ -2451,7 +2451,12 @@ Functions.checkPassword = function ($theForm) {
     var alertMessage = false;
 
     if ($password.val() === '') {
-        alertMessage = Messages.strPasswordEmpty;
+        alertMessage = Messages.strAllowNoPassowrdIsFalse;
+         if(window.confirm(alertMessage)){
+             alertMessage=false;
+        }else{
+            alertMessage = PMA_messages.strPasswordEmpty;        
+        }
```
Comment (williamdes) [inline js/src/functions.js:2459]: I think this is not the proper way to do this, @MauricioFauth do you have some advice ?

You should run `yarn js-lint --fix` on this block of code

**Comment by MauricioFauth** [comment]
Hi @RamiRima, thanks for your contribution.

As described in the issue, we should check if [`$GLOBALS['cfg']['Server']['AllowNoPassword'] === false`](https://docs.phpmyadmin.net/en/latest/config.html#cfg_Servers_AllowNoPassword) before warning the user.

Also, it's a best practice to use a different branch to open a pull request, this allows you to open more than one pull request.

**Pull request closed without merging**